003_data_logging/003_data_logging.tex

```diff
@@ -0,0 +1,86 @@
+
+First, it was important to enable an effective parametrisation of the simulation. A configuration interface was therefore set to abstract variables from the simulation environment. A configuration object would therefore be passed to the simulation environment instance, and it would define the following simulation parameters:
+
+\begin{multicols}{2}
+    \begin{enumerate}
+        \item Agent count for each team
+        \item Agent count for random agents
+        \item Agent count for selfish agents
+        \item Agent HP
+        \item Agents per floor
+        \item Amount of food on the platform
+        \item Food per agent ratio (optional)
+        \item Ticks per floor
+        \item Total days
+        \item Reshuffle period
+    \end{enumerate}
+\end{multicols}
+
+The configuration parameters could then be written in a simple \texttt{json} file which would be parsed into a configuration object.
+
+\section{General log dumps}
+
+Following the configuration work, a general logger was setup. The general logger is basically a single shared logger between the different instances in the system (simulation, tower, agents, etc...). Each instance appends by default an identifier to a log line in the general logger log file (named \texttt{main.json}). Here is an example below:
+
+\begin{verbatim}
+{"agent_id":"dece1252","agent_type":"Team5","floor":3,"health":100,
+"level":"info","msg":"Reporting agent state of team 5 agent","reporter":"agent"}
+
+{"Floor:":7,"HP:":100,"Social motive:":"Altruist","agent_id":"2bfd3cfb",
+"agent_type":"Team6","level":"info","msg":"Reporting agent state:","reporter":"agent"}
+
+{"agent_id":"7873bb50","agent_type":"Team1","floor":5,"health":100,
+"level":"info","msg":"Reporting agent state","reporter":"agent"}
+\end{verbatim}
+
+The general log dumps are not structured yet they enable agent teams to directly log custom states from their agents and use those in e.g. evolutionary algorithms or machine learning training.
+
+\section{Structured logs}
+
+The general log generates huge amounts of data and therefore was ineffective for data analysis and drawing conclusions easily from different simulations. Therefore, structured logs had to be introduced, where particular data is captured optimally to generate lightweight, understandable logs. Following consultations with the agent teams and relating as fit to the problem, the following structured logs where included:
+
+\begin{enumerate}
+    \item \textbf{Food}: food available on the platform and food taken by agents.
+    \item \textbf{Death}: agent death (including agent state at death time)
+    \item \textbf{Messages}: receiver, sender, message type and content
+    \item \textbf{Utility}: agent utility (including agent state)
+    \item \textbf{Story}: a combination of the above logs in a sequential manner
+\end{enumerate}
+
+The agent state encapsulates by default the agent age, floor, HP, utility, and custom state (e.g. emotions or memory).
```

Comment:
```suggestion
The agent state encapsulates by default the agent's age, floor, HP, utility, and custom state (e.g., emotions or memory).
```

Reply:
Do we need the comma in `(e.g., emotions or memory).` or typo?

Reply:
we do - agreed on slack last night